src/adapter/src/catalog/open.rs

```diff
@@ -189,6 +189,41 @@ impl Catalog {
         assert!(!updates.is_empty(), "initial catalog snapshot is missing");
         let mut txn = storage.transaction().await?;
 
+        // Force cc sizes to use swap, if requested.
+        if FORCE_SWAP_FOR_CC_SIZES.get(state.system_configuration.dyncfgs()) {
+            info!("force-enabling swap for cc replica sizes");
+
+            for size in state.cluster_replica_sizes.0.values_mut() {
+                if size.is_cc {
+                    size.swap_enabled = true;
+                    size.cpu_exclusive = false;
+                    size.selectors.remove("materialize.cloud/scratch-fs");
+                    size.selectors
+                        .insert("materialize.cloud/swap".into(), "true".into());
+                }
+            }
+
+            // The size definitions were copied into the in-memory replica configs during
+            // `CatalogState` initialization above. We can't move this code before that because we
+            // need to wait for the `system_configuration` to be fully initialized. Instead we also
+            // patch the replica configs here.
```

Comment:
Could you please also extend the comment with the new constraint that this has to be before the builtin table updates?